dask/dataframe/io/tests/test_parquet.py

```diff
@@ -4557,6 +4557,47 @@ def test_pyarrow_filesystem_option(tmp_path, fs):
     assert_eq(ddf, df)
 
 
+@PYARROW_MARK
+def test_fsspec_to_parquet_filesystem_option(tmp_path):
+    from fsspec import get_filesystem_class
+    from pyarrow import dataset as pa_da
+
+    key1 = "/read1"
+    key2 = str(tmp_path / "write1")
+
+    df = pd.DataFrame({"a": range(10)})
+    fs = get_filesystem_class("memory")(use_instance_cache=False)
+    pa_da.write_dataset(
+        pa.Table.from_pandas(df),
+        key1,
+        format="parquet",
+        filesystem=fs,
+    )
+
+    # read in prepared data
+    ddf = dd.read_parquet(
+        key1,
+        engine="pyarrow",
+        filesystem=fs,
+    )
+    ddf.to_parquet(key2, engine="pyarrow", filesystem=fs)
+
+    # make sure we didn't write to local fs
+    assert len(list(tmp_path.iterdir())) == 0, "wrote to local fs"
+
+    # make sure we wrote a key to memory fs
+    assert len(fs.ls(key2, detail=False)) == 1
+
+    # ensure append functionality works
+    ddf.to_parquet(key2, engine="pyarrow", append=True, filesystem=fs)
+    assert len(fs.ls(key2, detail=False)) == 2, "should have two parts"
+
+    rddf = dd.read_parquet(key2, engine="fastparquet", filesystem=fs)
+    assert_eq(rddf.partitions[0], ddf)
+    assert_eq(rddf.partitions[1], ddf)
+    pd.testing.assert_frame_equal(rddf.partitions[0].compute(), df)
```

Comment:
```suggestion
    assert_eq(ddf, df)
```
If this check passes (along with the two assertions above), then we know both partitions are the same as `df` after `compute`. Note that this assertion can also be moved to the like just before `ddf.to_parquet` for clarity.

Reply:
Note that you can also replace

```python
    assert_eq(rddf.partitions[0], ddf)
    assert_eq(rddf.partitions[1], ddf)
```
with 

```python
    assert_eq(rddf, dd.concat([ddf, ddf]))
```

Reply:
That makes sense. I wasn't sure whether assert_eq only compares the dask graphs, or actually evaluates the contents.